### ISSUE TITLE
Allow SENTs to change return value of CBaseEntity::PhysicsSolidMaskForEntity()

### ISSUE THREAD

**Issue by Jcw87** [Facepunch/garrysmod-requests#208]
It would be very useful if scripted entities could define their own CONTENTS masks. For example, I want to have an entity block players and physics objects, but not bullets. This can only be achieved by changing the return value of CBaseEntity::PhysicsSolidMaskForEntity().

There used to be a function to do this, but it was removed in some update, and I can't remember what it was called before.

**Comment by edgarasf123** [comment]
Bump, this request is 3 years old. And I'm looking for this feature.

**Comment by edgarasf123** [comment]
https://github.com/Facepunch/garrysmod-requests/issues/1055 This in a way allows you to "set" content flags.

**Comment by robotboy655** [comment]
no, only retrieve them

**Comment by edgarasf123** [comment]
Doesn't returning nothing makes the rays to go pass through the entity?
```lua
local sent_contents_flags = CONTENTS_PLAYERCLIP + CONTENTS_MONSTERCLIP

function ENT:TestCollision( startpos, delta, isbox, extents, mask )
	if bit.band( mask, sent_contents_flags ) ~= 0 then
		return {}
	end
	return nil
end
```

**Comment by edgarasf123** [comment]
Yeah, my alternative solution didn't work after the update. I really hoped it did. You need to return new hit pos which is bit hard to calculate.

**Comment by edgarasf123** [comment]
I believe this got implemented in next update? 

Can anyone confirm this also work with [Entity:PhysicsInitConvex](http://wiki.garrysmod.com/page/Entity/PhysicsInitConvex)?

![](https://i.imgur.com/SeK1hiB.png)

**Comment by Kefta** [comment]
It works with all physics objects.

**Comment by Zet0rz** [comment]
> I believe this got implemented in next update?

Setting the physics object to for example `CONTENTS_EMPTY` will make props pass through properly, but traces still hit, meaning players, nextbots, bullets, and lua traces will still collide with the entity regardless of what the physics object's Contents is. `util.PointContents` will also still return the default value, `CONTENTS_SOLID` for the entity/model I tested with.

> Can anyone confirm this also work with [Entity:PhysicsInitConvex](http://wiki.garrysmod.com/page/Entity/PhysicsInitConvex)?

Appears to have no effect on that either, beyond props being able to pass through.

Exposing this function would be very useful for creating things like props that only block players or NPCs  using the various clip enums.

**Comment by robotboy655** [comment]
Yeah this hook unfortunately doesn't do much, the coolest thing it can is making entities collide with NPCCLIP brushes.

**Comment by Kefta** [comment]
I thought the whole point of this hook was for changing interactions with brushes/player movement. https://github.com/Facepunch/garrysmod-requests/issues/1074 and https://github.com/Facepunch/garrysmod-requests/issues/1085 would be more useful though.
